modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/passthru/TargetResponseTest.java

```diff
@@ -0,0 +1,148 @@
+@PrepareForTest({ TargetContext.class })
+public class TargetResponseTest extends TestCase {
+    private static Log logger = LogFactory.getLog(TargetResponseTest.class.getName());
+
+    /**
+     * Testing the starting of target response when response body is not expected
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFalse() throws Exception {
+        ConfigurationContext configurationContext = new ConfigurationContext(new AxisConfiguration());
+        WorkerPool workerPool = new NativeWorkerPool(3, 4, 5, 5, "name", "id");
+        PassThroughTransportMetricsCollector metrics = new PassThroughTransportMetricsCollector(true, "testScheme");
+
+        TargetConfiguration targetConfiguration = new TargetConfiguration(configurationContext, null, workerPool,
+                metrics, null);
+        HttpResponse response = PowerMockito.mock(HttpResponse.class, Mockito.RETURNS_DEEP_STUBS);
+        NHttpClientConnection conn = PowerMockito.mock(NHttpClientConnection.class, Mockito.RETURNS_DEEP_STUBS);
+        TargetConnections connections = PowerMockito.mock(TargetConnections.class);
+        targetConfiguration.setConnections(connections);
+
+        PowerMockito.mockStatic(TargetContext.class);
+
+        TargetResponse targetResponse = new TargetResponse(targetConfiguration, response, conn, false, false);
+
+        try {
+            targetResponse.start(conn);
+        } catch (Exception e) {
+            logger.error(e);
+            Assert.fail("Unable to start the target response!");
+        }
+    }
+
+    /**
+     * Testing the starting of target response when response body is expected
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testTrue() throws Exception {
+        ConfigurationContext configurationContext = new ConfigurationContext(new AxisConfiguration());
+        WorkerPool workerPool = new NativeWorkerPool(3, 4, 5, 5, "name", "id");
+        PassThroughTransportMetricsCollector metrics = new PassThroughTransportMetricsCollector(true, "testScheme");
+
+        TargetConfiguration targetConfiguration = new TargetConfiguration(configurationContext, null, workerPool,
+                metrics, null);
+        targetConfiguration.build();
+        HttpResponse response = PowerMockito.mock(HttpResponse.class, Mockito.RETURNS_DEEP_STUBS);
+        NHttpClientConnection conn = PowerMockito.mock(NHttpClientConnection.class, Mockito.RETURNS_DEEP_STUBS);
+
+        PowerMockito.mockStatic(TargetContext.class);
+
+        TargetContext cntxt = new TargetContext(targetConfiguration);
+        PowerMockito.when(TargetContext.get(any(NHttpClientConnection.class))).thenReturn(cntxt);
+
+        TargetResponse targetResponse = new TargetResponse(targetConfiguration, response, conn, true, false);
+
+        try {
+            targetResponse.start(conn);
+        } catch (Exception e) {
+            logger.error(e);
+            Assert.fail("Unable to start the target response!");
```

Comment:
This test checks whether a connection with hard-coded params can be established to start reading from the target. Since there's no real target, we don't actually get any response. Therefore, we cannot assert based on getStatus.